### DISCUSSION TITLE
Why are comments only allowed in certain places?

### DISCUSSION THREAD

**Discussion by pingiun** [kdl-org/kdl#270] [General]
When writing my implementation I was quite confused that a comment like this is not allowed:

```kdl
node key=(type/*huh*/)10
```

Why not? it seems like commenting something about a type could be useful. In fact any place where no whitespace is allowed, no comments are allowed either. It seems like comments should be allowed between any token.

**Comment by Patitotective** [comment]
I'm also curious about this.

**Comment by zkat** [comment]
yeah you're not the first one to bring up whitespace (and thus, comments) inside typespecs.

I guess honestly it just didn't come up, and being more restrictive seemed like a great idea. It's something we can discuss as part of 2.0, though?

**Comment by LemmaEOF** [comment]
This kinda extends to a more general question about whitespace - where should it be allowed or disallowed? Should `foo bar = baz` be legal? What about `foo{bar;baz;qux;}`? I'm personally of the opinion that that you should be able to minimize or maximize KDL syntax as much as you'd like within a single line, but that might not be the same opinion as the team.

**Comment by zkat** [comment]
I've made issues for this for kdl 2.0:

* https://github.com/kdl-org/kdl/issues/342
* https://github.com/kdl-org/kdl/issues/341
